src/main/java/org/candlepin/subscriptions/tally/InventoryAccountUsageCollector.java

```diff
@@ -115,7 +115,15 @@ public InventoryAccountUsageCollector(FactNormalizer factNormalizer,
                     accountCalc.setOwner(owner);
                 }
 
-                Host host = new Host(hostFacts, facts);
+                Host existingHost = inventoryHostMap.remove(hostFacts.getInventoryId().toString());
+                Host host = existingHost == null ? new Host(hostFacts, facts) : existingHost;
+                if (existingHost != null) {
+                    host.getBuckets().clear(); // ensure we recalculate to remove any stale buckets
+                }
+                else {
+                    host.populateFieldsFromHbi(hostFacts, facts);
+                }
```

Comment:
The old code was
```java
    Host existingHost = inventoryHostMap.remove(hostFacts.getInventoryId().toString());
    Host host;
    if (existingHost == null) {
        host = new Host(hostFacts, facts);
    }
    else {
        host = existingHost;
        host.getBuckets().clear(); // ensure we recalculate to remove any stale buckets
        host.populateFieldsFromHbi(hostFacts, facts);
    }
    return host;    
```

I'm not sure that this new code is equivalent.  Specifically, in the old code if `existingHost` was non-null, `populateFieldsFromHbi` would run.  In the new code, it's only run if `existingHost` is null.

Maybe I am misreading because it is close to lunch and I am hungry.  I wouldn't be opposed to just dropping in the old code inline here (but I would add a comment above the assignment to `existingHost` highlighting that the map is being mutated).

Reply:
@awood thanks, my refactor was just bad. I did a fixup (see 2f18d7f), and rebased while I was at it, due to test failures.